buildSrc/src/main/scala/mesosphere/gradle/aws/S3Upload.scala

```diff
@@ -1,53 +1,66 @@
 package mesosphere.gradle.aws
 
-import java.io.File
+import java.util.concurrent.CompletableFuture
 
-import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
-import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
-import com.amazonaws.services.s3.transfer.Transfer.TransferState
-import com.amazonaws.services.s3.transfer.TransferManagerBuilder
+import org.apache.commons.io.FilenameUtils
 import com.typesafe.scalalogging.StrictLogging
 import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileTree
 import org.gradle.api.tasks.TaskAction
+import software.amazon.awssdk.core.async.AsyncRequestBody
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.S3AsyncClient
+import software.amazon.awssdk.services.s3.model.{PutObjectRequest, PutObjectResponse}
 
 import scala.beans.BeanProperty
 
 class S3Upload extends DefaultTask with StrictLogging {
 
-  val credentialProviderChain = new DefaultAWSCredentialsProviderChain()
+  lazy val s3: S3AsyncClient = S3AsyncClient.builder().region(Region.of(region)).build()
 
   @BeanProperty var region: String = "us-west-2"
 
   @BeanProperty var bucket: String = ""
 
   @BeanProperty var prefix: String = ""
```

Comment:
what kind of prefix is this? The name is very generic and the default value is "" so 🤔 maybe add some docs or rename?

Reply:
Good point. The AWS SDK 1.x hat this. I'll add docs.